src/subscription-user/subscription-user.controller.ts

```diff
@@ -28,4 +38,27 @@ export class SubscriptionUserController {
       createSubscriptionUserDto,
     );
   }
+
+  @Patch('/whatsapp')
+  @ApiBearerAuth('access-token')
+  @ApiOperation({
+    description: 'Cancel an active whatsapp subscription',
+  })
+  @UseGuards(FirebaseAuthGuard)
+  async cancelWhatsappSubscription(
+    @Req() req: Request,
+    @Body() updateSubscriptionsDto: UpdateSubscriptionUserDto,
+  ) {
+    // This endpoint cannot be used to activate a subscription, it can only be used to cancel.
+    if (updateSubscriptionsDto.isActive) {
+      throw new HttpException(
```

Comment:
Same thought as above - if this is .isActive =true could you reactivate a subscription?

Reply:
I could - and then add a call to respond.io to add the contact - but it seemed over complicated.